src/components/Form.js

```diff
@@ -1,55 +1,168 @@
 import styled from 'styled-components';
 import Button from './Button';
+import { useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router';
+import { useEffect } from 'react';
 
 export default function Form({ onAddActivity }) {
-  function handleSubmit(event) {
-    event.preventDefault();
-    const form = event.target;
-    const { activity, friend } = form.elements;
-    onAddActivity({ activity: activity.value, friend: friend.value });
-    activity.value = '';
-    friend.value = '';
-  }
+  const navigate = useNavigate();
+  const {
+    register,
+    handleSubmit,
+    setFocus,
+    formState: { errors },
+  } = useForm();
+
+  const onSubmit = data => {
+    onAddActivity(data);
+    navigate('/');
+  };
+
+  useEffect(() => {
+    setFocus('activity');
+  }, [setFocus]);
 
   return (
     <WrapperForm
       title="add activities"
       autoComplete="off"
-      onSubmit={handleSubmit}
+      onSubmit={handleSubmit(onSubmit)}
     >
-      <label htmlFor="activity">name of activity:</label>
-      <input id="activity" required type="text" name="activity" />
-      <label htmlFor="friend">who should join you?</label>
-      <input id="friend" type="text" name="friend" />
-      <PositionButton type="submit">Add</PositionButton>
+      <label htmlFor="activity">
+        name of activity:{' '}
+        <input
+          id="activity"
+          type="text"
+          name="activity"
+          {...register('activity', {
+            required: 'So, you plan to do nothing?? 😉',
+            maxLength: { value: 50 },
+            minLength: {
+              value: 3,
+              message:
+                'This is an activity with not even 2 characters? - I do not believe you..',
+            },
+          })}
+        />
+        {errors.activity && (
+          <ErrorMessage name="error-message">
+            {errors.activity.message}
+          </ErrorMessage>
+        )}
+      </label>
+
+      <label htmlFor="friend">
+        who should join you?{' '}
+        <input
+          id="friend"
+          type="text"
+          name="friend"
+          {...register('friend', {
+            maxLength: {
+              value: 100,
+              message: 'I can not believe that someone has so many friends',
+            },
+          })}
+        />
+        {errors.friend && (
+          <ErrorMessage name="error-message">
+            {errors.friend.message}
+          </ErrorMessage>
+        )}
+      </label>
+
+      <label htmlFor="notes">
+        space for some additional notes...
+        <textarea
+          id="notes"
+          type="text"
+          name="notes"
+          row="6"
```

Comment:
Don't know if it's right, but I prefer to style the textarea completely with css and not predefine col and rows